internal/zinc-core/src/main/scala/sbt/internal/inc/Incremental.scala

```diff
@@ -1068,6 +1068,25 @@ private final class AnalysisCallback(
     }
   }
 
+  def getSourceInfos: SourceInfos = {
+    // Collect Source Info from current run
+    val sources = reporteds.keySet ++ unreporteds.keySet ++ mainClasses.keySet
+    val sourceToInfo = sources.map { source =>
```

Comment:
Issue: I did some more manual testing and found out that `reported` isn't always complete. When one of the recompiled file has compilation error, warnings and errors from other recompiled files may or may not get reported by the compiler...